src/app/ui-components/sidebar-right/sidebar-right.component.html

```diff
@@ -1,32 +1,123 @@
+                                type="button"
+                                class="btn"
+                                id="dropdownbutton-{{sidebarElement.getName()}}"
+                                ngbDropdownToggle
+                                (hidden)="removeCssClass(sidebarElement.getUUID())"
+                                (shown)="addCssClass(sidebarElement.getUUID())">
+                                <div class="dot ml-1"></div>
+                                <div class="dot ml-1"></div>
+                                <div class="dot ml-1"></div>
+                            </button>
+                            <div
+                                ngbDropdownMenu
+                                attr.aria-labelledby="dropdownbutton-{{sidebarElement.getName()}}">
+                                <button
+                                    id="edit"
+                                    ngbDropdownItem
+                                    (click)="calbackMethods[1].clickCallback(sidebarElement.getUUID())">
+                                    <img
+                                        class="item pr-1 mb-1"
+                                        *ngIf="calbackMethods[1].icon !== ''"
+                                        [src]="calbackMethods[1].icon" />
+                                    <label>{{calbackMethods[1].label}}</label>
+                                </button>
+                                <div *ngIf="lStorage == 'chat'">
+                                    <button
+                                        id="export"
+                                        ngbDropdownItem
+                                        (click)="calbackMethods[3].clickCallback(sidebarElement.getUUID())">
+                                        <img
+                                            class="item pr-1 mb-2"
+                                            *ngIf="calbackMethods[3].icon !== ''"
+                                            [src]="calbackMethods[3].icon" />
+                                        <label
+                                            >{{calbackMethods[3].label}}</label
+                                        >
+                                    </button>
+                                </div>
+                                <button
+                                    id="delete"
+                                    ngbDropdownItem
+                                    (click)="calbackMethods[2].clickCallback(sidebarElement.getUUID())">
+                                    <img
+                                        class="item pr-1 mb-1"
+                                        *ngIf="calbackMethods[2].icon !== ''"
+                                        [src]="calbackMethods[2].icon" />
+                                    <label>{{calbackMethods[2].label}}</label>
+                                </button>
+                            </div>
```

Comment:
can't this be replaced with a '*ngFor' (ids could be inferred from the label or simply be another field of the callbackMethod)

Reply:
resolved